src/core/lv_refr.c

```diff
@@ -964,5 +964,16 @@ static void call_flush_cb(lv_disp_drv_t * drv, const lv_area_t * area, lv_color_
 {
     TRACE_REFR("Calling flush_cb on (%d;%d)(%d;%d) area with %p image pointer", area->x1, area->y1, area->x2, area->y2,
                color_p);
-    drv->flush_cb(drv, area, color_p);
+
+    lv_coord_t hor_res = drv->display_hor_res > 0 ? drv->display_hor_res : drv->hor_res;
+    lv_coord_t ver_res = drv->display_ver_res > 0 ? drv->display_ver_res : drv->ver_res;
+
+    lv_area_t offset_area = {
+        .x1 = LV_MAX(0, LV_MIN(area->x1 + drv->display_offset_x, hor_res - 1)),
+        .y1 = LV_MAX(0, LV_MIN(area->y1 + drv->display_offset_y, ver_res - 1)),
+        .x2 = LV_MAX(0, LV_MIN(area->x2 + drv->display_offset_x, hor_res - 1)),
```

Comment:
At this stage we can't truncate the area because `color_p` still contains `drv->hor_res` lines. So if `area` is different than `color_p` was rendered width to user won't be able to find the original dimensions of `color_p`.

Area should be truncated much earlier in the pipeline, but let's postpone this problem a little :slightly_smiling_face: 

Here I suggest only moving the area by an offset and ignore that case if it's out of the display.

Reply:
Ah, great catch, I missed this. :facepalm: 

Changed it to only apply the offset here without any truncation.